packages/core/src/interpreter.ts

```diff
@@ -204,8 +205,10 @@ export class Interpreter<
       deferredFn(state);
     }
 
-    for (const observer of this.observers) {
-      observer.next?.(snapshot);
+    if (previousSnapshot !== snapshot) {
+      for (const observer of this.observers) {
+        observer.next?.(snapshot);
+      }
     }
```

Comment:
that's the core of the change really but I had to remove mutations from actors to make this check work since with mutations... the snapshots **were** the same, they were just mutated ;p